src/us/kbase/workspace/database/provenance/CreditMetadata.java

```diff
@@ -0,0 +1,479 @@
+				&& Objects.equals(titles, other.titles)
+				&& Objects.equals(version, other.version);
+	}
+
+	/**
+	 * Gets a builder for the {@link CreditMetadata}.
+	 *
+	 * @param identifier
+	 *                unique persistent ID for the resource (i.e. the source data for this
+	 *                workspace object). Should be in the format
+	 *                <database name>:<identifier within database>
+	 * @param resourceType
+	 *                type of the resource, as a {@link ResourceType}
+	 * @param contributors
+	 *                list of {@link Contributor} objects; at least one contributor is required
+	 *                Duplicate and null entries are removed from the list.
+	 * @param titles
+	 *                list of {@link Title} objects; at least one title is required.
+	 *                Duplicate and null entries are removed from the list.
+	 * @return the builder.
+	 */
+	public static Builder getBuilder(
+			final String identifier,
+			final ResourceType resourceType,
+			final List<Contributor> contributors,
+			final List<Title> titles) {
+		final List<String> errorList = new ArrayList<>();
+		if (resourceType == null) {
+			errorList.add("resourceType cannot be null");
+		}
+		return new Builder(identifier, resourceType, contributors, titles,
+				errorList);
+	}
+
+	/**
+	 * Gets a builder for the {@link CreditMetadata}.
+	 *
+	 * @param identifier
+	 *                unique persistent ID for the resource (i.e. the source data for this
+	 *                workspace object). Should be in the format
+	 *                <database name>:<identifier within database>
+	 * @param resourceType
+	 *                type of the resource as a string
+	 * @param contributors
+	 *                list of {@link Contributor} objects; at least one contributor is required.
+	 *                Duplicate and null entries are removed from the list.
+	 * @param titles
+	 *                list of {@link Title} objects; at least one title is required
+	 *                Duplicate and null entries are removed from the list.
+	 * @return the builder.
+	 */
+	public static Builder getBuilder(
+			final String identifier,
+			final String resourceType,
+			final List<Contributor> contributors,
+			final List<Title> titles) {
+		final List<String> errorList = new ArrayList<>();
+
+		ResourceType rt = null;
+		if (resourceType == null || resourceType.isBlank()) {
+			errorList.add("resourceType cannot be null or whitespace only");
+		}
+		else {
+			try {
+				rt = ResourceType.getResourceType(resourceType);
+			} catch (IllegalArgumentException e) {
+				errorList.add(e.getMessage());
+			}
+		}
+		return new Builder(identifier, rt, contributors, titles, errorList);
+	}
+
+
+	// Regular expression used to detect URL-like strings; detects
+	// a sequence of letters at the start of a line, followed by one or more
+	// colons and one or more slashes.
+	static final Pattern pattern = Pattern.compile("^[a-zA-Z]+:+/+");
```

Comment:
I would make this private just so the next person that comes along doesn't waste time trying to figure out why it isn't when it can be

Reply:
:+1: